### PR TITLE
Add copy-snippet button in form

### LINKED ISSUE #6609: Option to copy snippets like pages.
# Problem description

We sometimes have snippets with a lot of fields that we would like to be able to copy as quickly as pages if needed.

image

### PR DESCRIPTION
**Pull request by wachterjohannes**
| Q | A
| --- | ---
| Bug fix? | no
| New feature? | yes
| BC breaks? | no
| Deprecations? | no
| Fixed tickets | fixes #6609
| Related issues/PRs | none
| License | MIT
| Documentation PR | none

#### What's in this PR?

This PR add the copy button to the snippet form

### CODE REVIEW

**Comment by wachterjohannes** [comment]
@alexander-schranz have extracted the php-cs fixes and phpstan fixes into an own PR https://github.com/sulu/sulu/pull/6862

**Comment by wachterjohannes** [comment]
@alexander-schranz ping

File: src/Sulu/Bundle/SnippetBundle/Admin/SnippetAdmin.php [hunk 1]
```diff
@@ -1,5 +1,7 @@
 <?php
 
+declare(strict_types=1);
+
```
Comment (alexander-schranz) [inline src/Sulu/Bundle/SnippetBundle/Admin/SnippetAdmin.php]: We hav eno strict_types yet in any files.
```suggestion
```

**Comment by wachterjohannes** [comment]
@alexander-schranz resolved

File: src/Sulu/Bundle/SnippetBundle/Admin/SnippetAdmin.php [hunk 2]
```diff
@@ -73,7 +76,7 @@ public function __construct(
         ViewBuilderFactoryInterface $viewBuilderFactory,
         SecurityCheckerInterface $securityChecker,
         WebspaceManagerInterface $webspaceManager,
-        $defaultEnabled
+        $defaultEnabled,
```
Comment (alexander-schranz) [inline src/Sulu/Bundle/SnippetBundle/Admin/SnippetAdmin.php]: we don't have this in the sulu php-cs yet

```suggestion
        $defaultEnabled
```

File: src/Sulu/Bundle/SnippetBundle/Admin/SnippetAdmin.php [hunk 3]
```diff
@@ -130,14 +151,14 @@ public function configureViews(ViewCollection $viewCollection): void
                     ->addLocales($snippetLocales)
                     ->setAddView(static::ADD_FORM_VIEW)
                     ->setEditView(static::EDIT_FORM_VIEW)
-                    ->addToolbarActions($listToolbarActions)
+                    ->addToolbarActions($listToolbarActions),
```
Comment (alexander-schranz) [inline src/Sulu/Bundle/SnippetBundle/Admin/SnippetAdmin.php]:
```suggestion
                    ->addToolbarActions($listToolbarActions)
```

File: src/Sulu/Bundle/SnippetBundle/Admin/SnippetAdmin.php [hunk 4]
```diff
@@ -130,14 +151,14 @@ public function configureViews(ViewCollection $viewCollection): void
...
             );
             $viewCollection->add(
                 $this->viewBuilderFactory
                     ->createResourceTabViewBuilder(static::ADD_FORM_VIEW, '/snippets/:locale/add')
                     ->setResourceKey(SnippetDocument::RESOURCE_KEY)
                     ->addLocales($snippetLocales)
-                    ->setBackView(static::LIST_VIEW)
+                    ->setBackView(static::LIST_VIEW),
```
Comment (alexander-schranz) [inline src/Sulu/Bundle/SnippetBundle/Admin/SnippetAdmin.php]:
```suggestion
                    ->setBackView(static::LIST_VIEW)
```

File: src/Sulu/Bundle/SnippetBundle/Admin/SnippetAdmin.php [hunk 5]
```diff
@@ -146,23 +167,23 @@ public function configureViews(ViewCollection $viewCollection): void
                     ->setTabTitle('sulu_admin.details')
                     ->setEditView(static::EDIT_FORM_VIEW)
                     ->addToolbarActions($formToolbarActionsWithType)
-                    ->setParent(static::ADD_FORM_VIEW)
+                    ->setParent(static::ADD_FORM_VIEW),
```
Comment (alexander-schranz) [inline src/Sulu/Bundle/SnippetBundle/Admin/SnippetAdmin.php]:
```suggestion
                    ->setParent(static::ADD_FORM_VIEW)
```

File: src/Sulu/Bundle/SnippetBundle/Admin/SnippetAdmin.php [hunk 6]
```diff
@@ -146,23 +167,23 @@ public function configureViews(ViewCollection $viewCollection): void
...
             );
             $viewCollection->add(
                 $this->viewBuilderFactory
                     ->createResourceTabViewBuilder(static::EDIT_FORM_VIEW, '/snippets/:locale/:id')
                     ->setResourceKey(SnippetDocument::RESOURCE_KEY)
                     ->addLocales($snippetLocales)
                     ->setBackView(static::LIST_VIEW)
-                    ->setTitleProperty('title')
+                    ->setTitleProperty('title'),
```
Comment (alexander-schranz) [inline src/Sulu/Bundle/SnippetBundle/Admin/SnippetAdmin.php]:
```suggestion
                    ->setTitleProperty('title')
```

File: src/Sulu/Bundle/SnippetBundle/Admin/SnippetAdmin.php [hunk 7]
```diff
@@ -146,23 +167,23 @@ public function configureViews(ViewCollection $viewCollection): void
...
             );
             $viewCollection->add(
                 $this->viewBuilderFactory->createFormViewBuilder('sulu_snippet.edit_form.details', '/details')
                     ->setResourceKey(SnippetDocument::RESOURCE_KEY)
                     ->setFormKey('snippet')
                     ->setTabTitle('sulu_admin.details')
                     ->addToolbarActions($formToolbarActionsWithType)
-                    ->setParent(static::EDIT_FORM_VIEW)
+                    ->setParent(static::EDIT_FORM_VIEW),
```
Comment (alexander-schranz) [inline src/Sulu/Bundle/SnippetBundle/Admin/SnippetAdmin.php]:
```suggestion
                    ->setParent(static::EDIT_FORM_VIEW)
```

File: src/Sulu/Bundle/SnippetBundle/Admin/SnippetAdmin.php [hunk 8]
```diff
@@ -172,7 +193,7 @@ public function configureViews(ViewCollection $viewCollection): void
                     ->setTabTitle('sulu_snippet.taxonomies')
                     ->addToolbarActions($formToolbarActionsWithoutType)
                     ->setTitleVisible(true)
-                    ->setParent(static::EDIT_FORM_VIEW)
+                    ->setParent(static::EDIT_FORM_VIEW),
```
Comment (alexander-schranz) [inline src/Sulu/Bundle/SnippetBundle/Admin/SnippetAdmin.php]:
```suggestion
                    ->setParent(static::EDIT_FORM_VIEW)
```

File: src/Sulu/Bundle/SnippetBundle/Admin/SnippetAdmin.php [hunk 9]
```diff
@@ -181,7 +202,7 @@ public function configureViews(ViewCollection $viewCollection): void
                     ->setOption('tabTitle', 'sulu_snippet.default_snippets')
                     ->setOption('tabOrder', 3072)
                     ->setParent(PageAdmin::WEBSPACE_TABS_VIEW)
-                    ->addRerenderAttribute('webspace')
+                    ->addRerenderAttribute('webspace'),
```
Comment (alexander-schranz) [inline src/Sulu/Bundle/SnippetBundle/Admin/SnippetAdmin.php]:
```suggestion
                    ->addRerenderAttribute('webspace')
```

File: src/Sulu/Bundle/SnippetBundle/Tests/Unit/Admin/SnippetAdminTest.php [hunk 10]
```diff
@@ -118,6 +130,11 @@ public function testConfigureViews($locales): void
                 new Toolbaraction('sulu_admin.save'),
                 new Toolbaraction('sulu_admin.type', ['sort_by' => 'title']),
                 new Toolbaraction('sulu_admin.delete'),
+                new DropdownToolbarAction(
+                    'sulu_admin.edit',
+                    'su-pen',
+                    $dropdownActions,
```
Comment (alexander-schranz) [inline src/Sulu/Bundle/SnippetBundle/Tests/Unit/Admin/SnippetAdminTest.php]:
```suggestion
                    $dropdownActions
```

File: src/Sulu/Bundle/SnippetBundle/Tests/Unit/Admin/SnippetAdminTest.php [hunk 11]
```diff
@@ -137,6 +154,11 @@ public function testConfigureViews($locales): void
...
```
Comment (alexander-schranz) [inline src/Sulu/Bundle/SnippetBundle/Tests/Unit/Admin/SnippetAdminTest.php]:
```suggestion
                    $dropdownActions
```

**Review by alexander-schranz — CHANGES_REQUESTED** [review]
unexpected code style changes. if we change code style we need todo it on 2.4 / 2.5 via php-cs-fixer

**Comment by wachterjohannes** [comment]
@alexander-schranz fixed

**Review by alexander-schranz — APPROVED** [review]

**Pull request merged**